internal/engine/live_update_build_and_deployer.go

```diff
@@ -264,7 +264,7 @@ func (lubad *LiveUpdateBuildAndDeployer) containerUpdaterForSpecs(specs []model.
 		return lubad.ecu
 	}
 
-	if shouldUseSynclet(lubad.updMode, lubad.env, lubad.runtime) {
+	if lubad.shouldUseSynclet(specs, stateSet) {
```

Comment:
(ugh i can't comment on the line of code I actually care about but...)

re: ln 259, we might want to throw an error if updMode == Synclet but none of the k8s targets have an eligible synclet. Seems like we'd rather raise here where we can control the messaging better then just let something more obscure go wrong further down the pipe.

Reply:
i think that's a good idea, but I also think that messaging that effectively would end up being extremely complex - there's no single codepath that has access to all the information you need

Reply:
i think it's baaaasically right here. if you move the `containerUpdaterForSpecs` call down past where we know we have live update targets, then INSIDE the `containerUpdaterForSpecs` call you know:
1. you have live update targets
2. user explicitly specified updateMode=synclet
3. you tried and failed to inject the synclet into this k8s target

That's the info you need, right?

Reply:
you might have other k8s targets that want the synclet, and this only has access to one manifest, see discussion here:
https://github.com/windmilleng/tilt/issues/2948